python/equistore-operations/equistore/operations/reduce_over_samples.py

```diff
@@ -76,31 +92,36 @@ def _reduce_over_samples_block(
 
     if remaining_samples is None:
         assert sample_names is not None
-        remaining_samples = [
-            s_name for s_name in block_samples.names if s_name not in sample_names
-        ]
+        remaining_samples_final: List[str] = []
+        for s_name in block_samples.names:
+            if s_name in sample_names:
+                continue
+            remaining_samples_final.append(s_name)
+    else:
+        remaining_samples_final = remaining_samples
```

Comment:
```suggestion
        remaining_samples_names = remaining_samples
```

if we need a separate variable here, this might make the content a bit clearer

Reply:
Done